.github/workflows/rust.yml

```
@@ -61,26 +57,19 @@ jobs:
           - build: stable
             os: ubuntu-latest
             rust: stable
+
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ matrix.rust }}
-          override: true
+        uses: actions/checkout@v4
 
       - name: Install ${{ matrix.rust }} toolchain
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ matrix.rust }}
-          override: true
 
       - name: Run tests
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --all --no-fail-fast
+        run: |
+          cargo test --workspace --no-fail-fast --doc
```

As I understand it, `--doc` will suppress all other tests, so I think we could only do this is a separate step. But I also think, that the default mode includes doctests, according to the cargo book:
https://doc.rust-lang.org/cargo/commands/cargo-test.html#target-selection

Also, I think the `--workspace` option is on by default, but I do find the wording in the cargo book very inconclusive:
https://doc.rust-lang.org/cargo/commands/cargo-test.html#package-selection

But generally, I would try this:
```suggestion
          cargo test --no-fail-fast
```

And then look through the CI output to double-check that doctests are running...